### PR TITLE
REVIEW ONLY: Use updated MCUboot port for Mbed OS

### PR DESCRIPTION
**Pull request by LDong-Arm**
## Intro
This updates Embedded Planet's MCUboot bootloader demo app to use updated port for Mbed OS (https://github.com/LDong-Arm/mcuboot/pull/1).
Tested on DISCO_L475VG_IOT01A (full update flow) and NRF52840_DK (booting an initial image).

**Note**: README has not been updated

## Usage
To use,
* Generate an RSA key:
    ```
    ./mcuboot/scripts/imgtool.py keygen -k signing-keys.pem -t rsa-2048
    ```
    and a C file
    ```
    ./mcuboot/scripts/imgtool.py getpub -k signing-keys.pem > signing_keys.c
    ```
* Compile, build and flash as usual.

### CODE REVIEW

File: default_bd.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ mbed::BlockDevice* get_secondary_bd(void) {
     // slice it into a smaller size, 1MB at the beginning
 
     mbed::BlockDevice* default_bd = mbed::BlockDevice::get_default_instance();
-    static mbed::SlicingBlockDevice sliced_bd(default_bd, 0x0, 0xDD000);
+    static mbed::SlicingBlockDevice sliced_bd(default_bd, 0x0, MCUBOOT_SLOT_SIZE);
```
Comment (evedon) [inline default_bd.cpp:81]: It's worth updating the comment just above to remove the explicit size of the first slot.

**Review by evedon — APPROVED** [review]
Only a minor comment. Looks good.

File: default_bd.cpp (hunk 1 above)
Comment (LDong-Arm) [reply default_bd.cpp:81]: Done

**Comment by LDong-Arm** [comment]
@evedon PR to Embedded Planet: https://github.com/AGlass0fMilk/mbed-mcuboot-demo/pull/5

**Pull request closed without merging**